compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12845,195 +12845,292 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    }
 
 
+bool canTransformOp(bool canTransformAdd, bool canTransformSub, TR::Node *node, TR::Simplifier *s)
+   {
+   if (!(canTransformAdd || canTransformSub))
+      {
+      if (s->trace())
+         traceMsg(s->comp(),
+                  "\nEliminating add/sub under order compare node n%dn failed due to overflow\n",
+                  node->getGlobalIndex());
+
+      return false;
+      }
+   return true;
+   }
+
+// Overflow/Underflow handling:
+// Must detect overflow without causing overflow:
+// Equation: add operation: x + Const1 < Const2 --> x < Const2 - Const1
+//          sub operation: x - Const1 < Const2 --> x < Const2 + Const1
+
+//Signed Constants may be negative, therefore underflow and overflow are possible for both add and sub operations
+bool isNotSignedOverflow(bool isAddOp, bool isSubOp, uint64_t oldConst1, uint64_t oldConst2, uint64_t signedMax){
```

Comment:
why the bool for add and the bool for sub?

Reply:
because I need to check for overflow in both an add op and a sub op
if add op: x+a>b will overflow at x> b-a if b-a overflows, when a is a very large negative number
if sub op: x-a> b will overflow at x>b+a if b+a overflows, when a is a very large positive number
Am I missing something?

Reply:
I was more meaning you don't need two bools - either you are checking an add or a sub. So either the name changes to isNotSignedOverflowAddSub or something or you pass in the opcode and assert it is either and isAdd or an isSub or whatever. The two bools are confusing.